mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -2271,6 +2328,289 @@ pub struct TakerSwapPreparedParams {
     maker_payment_spend_trade_fee: TradeFee,
 }
 
+pub async fn check_watcher_payments(
+    swap: &TakerSwap,
+    ctx: &MmArc,
+    mut saved: TakerSavedSwap,
+    command: Option<TakerSwapCommand>,
+) -> Result<Option<TakerSwapCommand>, String> {
```

Comment:
Done

Reply:
> I also think that this function and the other related functions can be moved to a seperate file since taker_swap is growing very large. This part can be done in next PRs, but it's good to start separating watchers related code from main swap code otherwise things will get very complicated in the future.

I see that you moved it to `swap_watchers` but as I understand the code there is executed in watchers nodes, but this function is executed on a taker node, that's why I suggested a separate file (`taker_restart.rs`, `taker_watchers.rs`, etc..).

Reply:
Done